### PR TITLE
feat(gateway): match systemd service name to gateway name

### PR DESCRIPTION
**Pull request by bernie-g**
## Summary
The systemd service name now matches the gateway name directly (e.g. `my-gateway.service` instead of the hardcoded `infisical-gateway.service`), with config at `/etc/infisical/gateways/<name>.conf`. Legacy installs are detected and updated in place with a warning to migrate. The `systemd uninstall` command now requires a gateway name argument.

Companion docs PR: https://github.com/Infisical/infisical/pull/6624

## Test plan
- [x] All install/uninstall edge cases tested in Docker with systemd

### CODE REVIEW

File: packages/gateway-v2/systemd.go [hunk 1]
```diff
@@ -11,20 +11,82 @@ import (
...
+	gatewaysConfigDir  = "/etc/infisical/gateways"
+)
+
+func serviceFilePath(name string) string {
+	return fmt.Sprintf("/etc/systemd/system/%s.service", name)
+}
+
+func gatewayConfigPath(name string) string {
+	return filepath.Join(gatewaysConfigDir, name+".conf")
+}
```
[the remark this answers is not in the document]
Comment (bernie-g) [reply packages/gateway-v2/systemd.go:27]: If you're running this command you already have root access

File: packages/gateway-v2/systemd.go [hunk 2]
```diff
@@ -11,20 +11,82 @@ import (
...
+
+	if legacy.exists && legacy.gatewayName == name {
+		return installResult{
+			serviceName: legacyServiceName,
+			configPath:  legacyConfigPath,
+			isLegacy:    true,
+		}
+	}
+
+	if legacy.exists {
```
Comment (saifsmailbox98) [inline packages/gateway-v2/systemd.go:68]: if the user selects `infisical-gateway` as the name and there is a legacy service for an existing gateway, it will indicate that this service `will be installed alongside it`, but will overwrite the legacy service file.

File: packages/gateway-v2/systemd.go [hunk 3]
```diff
@@ -11,20 +11,82 @@ import (
...
+	if _, err := os.Stat(legacyServicePath); os.IsNotExist(err) {
+		return legacyInfo{}
+	}
+
+	name, _ := readKeyFromConfFile(legacyConfigPath, GATEWAY_NAME_ENV_NAME)
+	return legacyInfo{exists: true, gatewayName: name}
+}
+
+func logLegacyWarning(svcName string) {
+	log.Warn().Msgf("Using legacy service name '%s'. To migrate to the new naming format, run: sudo infisical gateway systemd uninstall %s && sudo infisical gateway systemd install %s <original-flags>", legacyServiceName, svcName, svcName)
```
Comment (saifsmailbox98) [inline packages/gateway-v2/systemd.go]: `<original-flags>` is not very helpful. we can ask them to get a new command/enrollment token from Infisical dashboard

File: packages/cmd/gateway.go [hunk 4]
```diff
@@ -675,38 +681,44 @@ var gatewaySystemdInstallCmd = &cobra.Command{
...
 var gatewaySystemdUninstallCmd = &cobra.Command{
-	Use:                   "uninstall",
+	Use:                   "uninstall [name]",
 	Short:                 "Uninstall and remove systemd service for the gateway (requires sudo)",
 	Long:                  "Uninstall and remove systemd service for the gateway. Must be run with sudo on Linux.",
-	Example:               "sudo infisical gateway systemd uninstall",
+	Example:               "sudo infisical gateway systemd uninstall my-gateway",
 	DisableFlagsInUseLine: true,
-	Args:                  cobra.NoArgs,
+	Args:                  cobra.ExactArgs(1),
```
Comment (saifsmailbox98) [inline packages/cmd/gateway.go]: `infisical gateway systemd uninstall` (wihtout any args) will break on CLI upgrade, not sure if someone might be doing it through scripts.

can we use `use cobra.MaximumNArgs(1)` instead? so uninstall legacy if present, else show error that name is needed?

**Review by saifsmailbox98 — CHANGES_REQUESTED** [review]

File: packages/cmd/gateway.go (hunk 4 above)
Comment (bernie-g) [reply packages/cmd/gateway.go]: I doubt anyone is running this in a script but yea we can do that

Comment (bernie-g) [reply packages/cmd/gateway.go]: Done

File: packages/gateway-v2/systemd.go (hunk 3 above)
Comment (bernie-g) [reply packages/gateway-v2/systemd.go]: Sure, done

File: packages/gateway-v2/systemd.go (hunk 2 above)
Comment (bernie-g) [reply packages/gateway-v2/systemd.go:68]: Good catch, fixed!

**Review by saifsmailbox98 — APPROVED** [review]
